examples/stm32/f4/stm32f429i-discovery/sdram/console.c

```diff
@@ -0,0 +1,235 @@
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2013 Chuck McManis <cmcmanis@mcmanis.com>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * USART example (alternate console)
+ *
+ * This version then adds in interrupts, which is really handy for
+ * the receive function as it is impossible to predict when someone
+ * might type a character, further you can create a "character based
+ * reset" capability if you choose to.
+ */
+
+#include <stdint.h>
+#include <setjmp.h>
+#include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/stm32/usart.h>
+#include <libopencm3/cm3/nvic.h>
+#include <libopencm3/stm32/iwdg.h>
+#include <libopencm3/cm3/scb.h>
+#include <libopencm3/cm3/cortex.h>
+#include "console.h"
+
+
+/*
+ * Some definitions of our console "functions" attached to the 
+ * USART.
+ *
+ * These define sort of the minimum "library" of functions which
+ * we can use on a serial port.
+ */
+
+
+#define CONSOLE_UART	USART1
+
+
+/* This is a ring buffer to holding characters as they are typed
+ * it maintains both the place to put the next character received
+ * from the UART, and the place where the last character was
+ * read by the program. See the README file for a discussion of
+ * the failure semantics.
+ */
+#define RECV_BUF_SIZE	128		// Arbitrary buffer size
+char recv_buf[RECV_BUF_SIZE];
+volatile int recv_ndx_nxt;		// Next place to store
+volatile int recv_ndx_cur;		// Next place to read
+
+/* For interrupt handling we add a new function which is called
+ * when recieve interrupts happen. The name (usart1_isr) is created
+ * by the irq.json file in libopencm3 calling this interrupt for
+ * USART2 'usart2', adding the suffix '_isr', and then weakly binding
+ * it to the 'do nothing' interrupt function in vec.c.
+ *
+ * By defining it in this file the linker will override that weak
+ * binding and instead bind it here, but you have to get the name
+ * right or it won't work. And you'll wonder where your interrupts
+ * are going.
+ */
+void usart1_isr(void) {
+	uint32_t	reg;
+	int			i;
+
+	do {
+		reg = USART_SR(CONSOLE_UART);
+		if (reg & USART_SR_RXNE) {
+			recv_buf[recv_ndx_nxt] = USART_DR(CONSOLE_UART);
+#ifdef RESET_ON_CTRLC
```

Comment:
This should be updated to the "current" code which includes scb.h and calls scb_system_reset() which works correctly now.